src/libs/HttpUtils.js

```diff
@@ -33,12 +40,26 @@ let cancellationController = new AbortController();
  * @returns {Promise}
  */
 function processHTTPRequest(url, method = 'get', body = null, canCancel = true) {
+    const startTime = new Date();
+
     return fetch(url, {
         // We hook requests to the same Controller signal, so we can cancel them all at once
         signal: canCancel ? cancellationController.signal : undefined,
         method,
         body,
     })
+        .then((response) => {
+            const match = url.match(regex)[1];
+            if (addSkewList.includes(match)) {
+                const serverTime = new Date(response.headers.get('Date'));
+                const endTime = new Date();
+                const latency = (endTime.valueOf() - startTime.valueOf()) / 2;
+                const skew = serverTime.valueOf() - startTime.valueOf() + latency;
```

Comment:
What if simplify calculations?
```suggestion
                const latency = (endTime - startTime) / 2;
                const skew = serverTime - startTime + latency;
```